SU-HSDO/suhumsci
# STN-730: Slider Styles

# READY FOR REVIEW

## Summary
The work in this PR adds unique styles for the indicator dots and toggles on gradient hero slides. Adjustments to spacing and the alignment of the gradient hero slider component were also addressed.

The arrow toggles on the Colorful theme use the primary color pairing, and on Traditional use global-black.
The bars that indicate which slide you are on use secondary on Colorful, and global-black on Traditional.

Slick slider arrow toggle and dots / bars styles have been placed into Sass mixins to allow for easy implementation on future hero slider work.

Designs: [Figma comps](https://www.figma.com/file/BieDQhj8BoLlHJrzpDaIXV/Slider-Container-for-Hero-Components?node-id=0%3A1)

This work has been:
- [x] Documented in the Color Pairing Spreadsheet
- [x] Browser Tested
- [x] Tested for accessibility
- [x] Tested for logged in / logged out states

## Need Review By (Date)
5/13/2021

## Urgency
Medium

## Steps to Test
1. In your CLI, run `npm run test` and confirm our tests pass.
2. If you have a flexible page filled with examples of gradient hero sliders, you may use that for testing. If not, create a flexible page and add a variety of gradient hero sliders. I recommend including examples of the following:
- [x] Gradient Hero Slider in the hero area at the top of a page
- [x] Gradient Hero Slider in the main content section
- [x] Gradient Hero Slider with slides that have their alignment set to the right
- [x] Gradient Hero Sliders with a combination of fields (title, copy, and button) present and with a variation of content
- [x] Gradient Hero Slider in a row
- [x] Gradient Hero Slider in a well in a row (Note that at small screen sizes, all heros are squished. I don't think this is a common use case.)
3. Confirm the arrow toggles and slide indicators (bars) match the gradient hero designs comp.
4. Confirm the arrow toggles and slide indicators do not interfere with caption and credits on images.
5. Confirm the arrow toggles are keyboard accessible. Note that when you are testing keyboard navigation, once you tab to an arrow toggle, on your keyboard use the arrow key to move through the slides. You will not be able to navigate through the slides with the keyboard using the slide indicator bars. Slick JS adds a role of `presentation` to the bars (dots) which overrides the link behavior. I also recommend testing the keyboard functionality in the logged out state to avoid the Drupal admin contextual links.
